### PR TITLE
chore(workspace-store): remove @scalar/code-highlight dependency

### PR DESCRIPTION
**Pull request by hanspagel**
Extracted from #8537

## Problem

too many dependencies, build takes too long

## Solution

get rid of a few dependencies

## Checklist

- [x] I explained why the change is needed.
- [ ] I added a changeset. 
- [ ] I added tests.
- [ ] I updated the documentation.

---

> [!NOTE]
> **Medium Risk**
> Moderate risk because markdown heading extraction behavior changes (new parser, new edge-case handling like setext/code-fence rules), which can affect navigation structure and generated slugs/IDs.
> 
> **Overview**
> **Removes `@scalar/code-highlight` from `@scalar/workspace-store`** by switching `getHeadingsFromMarkdown` to use a new `@scalar/helpers` utility (`getMarkdownHeadings`).
> 
> **Adds `getMarkdownHeadings` to `@scalar/helpers`** (new `markdown` export surface) with tests/benchmarks, covering ATX + setext headings, skipping fenced/indented code blocks, and stripping common inline markdown/HTML from heading text.
> 
> Updates build/packaging metadata (`helpers` exports + `sideEffects: false`, excludes `*.bench.ts` from build) and adds a `workspace-store` test to ensure setext headings appear in description navigation.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 22fbce1c91be25cbd4f195c4a90c162a97c1fb81. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/workspace-store/package.json [hunk 1]
```diff
@@ -139,23 +139,22 @@
...
     "@scalar/typebox": "catalog:*",
     "@scalar/types": "workspace:*",
     "github-slugger": "catalog:*",
-    "type-fest": "catalog:*",
     "vue": "catalog:*",
     "yaml": "catalog:*"
   },
   "devDependencies": {
     "@google-cloud/storage": "catalog:*",
+    "@scalar/snippetz": "workspace:*",
```
[the remark this answers is not in the document]
Comment (hanspagel) [reply packages/workspace-store/package.json]: what is old-definitions 🤔
